### PR TITLE
Add xs2a submitAuthorizationCode implementation

### PR DESCRIPTION
**Pull request by artemsnisarenko**

### CODE REVIEW

File: onlinebanking-xs2a/src/main/java/de/adorsys/xs2a/XS2ABanking.java [hunk 1]
```diff
@@ -343,7 +343,36 @@ private SelectPsuAuthenticationMethod buildSelectPsuAuthenticationMethod(Transac
 
     @Override
     public String submitAuthorizationCode(SubmitAuthorizationCodeRequest submitPaymentRequest) {
-        return null;
+        Xs2aTanSubmit tanSubmit = (Xs2aTanSubmit) submitPaymentRequest.getTanSubmit();
+        String bankingUrl2 = tanSubmit.getBankingUrl();
```
Comment (tadschik) [inline onlinebanking-xs2a/src/main/java/de/adorsys/xs2a/XS2ABanking.java]: -bankingUrl2 + bankingUrl

Comment (artemsnisarenko) [reply onlinebanking-xs2a/src/main/java/de/adorsys/xs2a/XS2ABanking.java]: Renamed.

File: onlinebanking-xs2a/src/main/java/de/adorsys/xs2a/XS2ABanking.java [hunk 2]
```diff
@@ -343,7 +343,36 @@ private SelectPsuAuthenticationMethod buildSelectPsuAuthenticationMethod(Transac
...
+        ApiClient apiClient = createApiClient(bankingUrl2);
+        PaymentInitiationServicePisApi pis = createPaymentInitiationServicePisApi(apiClient);
+
+        String tan = submitPaymentRequest.getTan();
+
+        String paymentId = tanSubmit.getPaymentId();
+        String authorisationId = tanSubmit.getAuthorisationId();
+
+        UUID xRequestId = UUID.randomUUID();
+
```
[the remark this answers is not in the document]
Comment (artemsnisarenko) [reply onlinebanking-xs2a/src/main/java/de/adorsys/xs2a/XS2ABanking.java]: Removed.

File: onlinebanking-xs2a/src/main/java/de/adorsys/xs2a/XS2ABanking.java [hunk 3]
```diff
@@ -343,7 +343,36 @@ private SelectPsuAuthenticationMethod buildSelectPsuAuthenticationMethod(Transac
...
+        try {
+            pis.updatePaymentPsuData(SINGLE_PAYMENT_SERVICE, SEPA_CREDIT_TRANSFERS, paymentId, authorisationId,
+                    xRequestId, transactionAuthorisation, null, null, null, psuId, null, psuCorporateId, null,
+                    PS_UIP_ADDRESS, null, null, null, null, null, null, null, null, null);
+        } catch (ApiException e) {
+            logger.error("Submit authorisation code failed", e);
+            throw new XS2AClientException(e);
+        }
+
+        return ""; // ?!
```
[the remark this answers is not in the document]
Comment (artemsnisarenko) [reply onlinebanking-xs2a/src/main/java/de/adorsys/xs2a/XS2ABanking.java]: Changed.

**Pull request merged**